Написаны тесты на сценарии, указанные в задании.

Vezaris opened the pull request:
Сгенерирован отчет Allure.
Добавлено описание в README.

Vezaris committed:
Написаны тесты на сценарии, указанные в задании.
Сгенерирован отчет Allure.
Добавлено описание в README.

pomegranate282922 commented on src/main/java/ru/yandex/practicum/Api/Model.java:
```diff
@@ -0,0 +1,56 @@
+import static io.restassured.RestAssured.given;
+
+public class Model {
+    private final Object OBJECT;
```
Нейминг - аналогично второму заданию 
Нужно исправить

pomegranate282922 commented on src/main/java/ru/yandex/practicum/Api/Model.java:
```diff
@@ -0,0 +1,56 @@
+    public void sendPostRegister() {
+        given()
+                .header("Content-type", "application/json")
+                .body(OBJECT)
```
Используем не Object, а конкретные классы 
Нужно исправить

pomegranate282922 commented on src/main/java/ru/yandex/practicum/Api/RegisterUserApi.java:
```diff
@@ -0,0 +1,37 @@
+package ru.yandex.practicum.Api;
```
Пакеты именуем строчными буквами без camelCase 
Нужно исправить

pomegranate282922 commented on src/main/java/ru/yandex/practicum/Page/AuthUserPage.java:
```diff
@@ -0,0 +1,74 @@
+    private final AuthUserApi AUTHUSERAPI;
+
+    /*---------------------------------- ЛОКАТОРЫ ------------------------*/
+    private final By MAIN_ENTRY_BUTTON = By.xpath("//button[text()='Войти в аккаунт']");
```
Нейминг 
Нужно исправить

pomegranate282922 commented on src/main/java/ru/yandex/practicum/Page/NavigationUserPage.java:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    @Step("Выход из личного кабинета")
+    public WebElement LogOut() {
```
Методы именуем начиная со строчной буквы 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/practicum/DriverRule.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    protected void before() {
+        if ("yandex".equals(System.getProperty("browser")))
```
Отлично что выбор браузера делается через сис переменную

pomegranate282922 commented on src/test/java/ru/yandex/practicum/NavigationUserTest.java:
```diff
@@ -0,0 +1,62 @@
+    @Test
+    @DisplayName("Выход из личного кабинета")
+    @Description("Успешный выход из личного кабинета")
+    public void LogOut() {
```
Нейминг нужно поправить везде

pomegranate282922 commented on src/test/java/ru/yandex/practicum/NavigationUserTest.java:
```diff
@@ -0,0 +1,62 @@
+    @DisplayName("Выбор раздела Соусы")
+    @Description("Раздел Соусы выбирается и происходит скрол до нужных товаров")
+    public void checkSauce() {
+        Assert.assertTrue(navigationUserPage.selectSauce().isDisplayed());
```
Здесь есть ошибочка 
ты проверяешь текст в разделах, но метод isDisplayed() (и Condition.visible) на самом деле считает, что текст виден, даже если он проскролен, т.к. этот метод проверяет что у локатора отсутствует свойство hidden.
Попробуй проскролить разделы и смотреть как меняются имена классов в верстке у разделов. И построй проверку "в каком разделе я нахожусь" на этой основе.
нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/practicum/BaseTest.java:
```diff
@@ -0,0 +1,32 @@
+    @Before
+    public void setUp () {
+        RestAssured.baseURI = ru.yandex.practicum.EnvConfig.URL_FOR_TEST_BURGER_API;
+        registerUserApi = new RegisterUserApi("myTestName", "myTestMail333@yandex.ru", "myTestPass333");
```
Лучше бы тоже рандомизировать 
Можно улучшить

pomegranate282922 reviewed (commented):
Хорошая работа! Все тесты реализованы. Применяется апи для вспомогательных шагов. 
замечания:
1. Нейминг - констант, методов, пакетов 
2. Ошибка в тестах на переходы 
3. Убрать Object 
4. Остальное по мелочи

Vezaris committed:
Исправлены замечания, а именно:
- поправлен нейминг пакетов, методов
- добавлен static для локаторов
- Object заменен на конкретный класс
- изменена проверка в тесте навигации по разделам конструктора (переход к булкам, соусам и т.д.)

pomegranate282922 reviewed (approved):
Отлично! Зачет!
